lib/event_sourcery/event_processing/event_stream_processor.rb

```diff
@@ -39,23 +45,40 @@ def process(event)
       end
 
       module ClassMethods
+
+        # @attr_reader processes_event_types [Array] Process Event Types
+        # @attr_reader event_handlers [Hash] Hash of handler blocks keyed by event
         attr_reader :processes_event_types, :event_handlers
 
+        # This will register the event types to process.
+        #
+        # @param event_types a collection of event types to process
         def processes_events(*event_types)
           @processes_event_types = Array(@processes_event_types) | event_types.map(&:to_s)
         end
 
+        # Process all the events.
```

Comment:
It seems so https://github.com/envato/event_sourcery/blob/9a991666d147f295f7a566fe40583d57dbcc017e/lib/event_sourcery/event_processing/event_stream_processor.rb#L38-L38